ProjectLighthouse/Pages/UserPage.cshtml

```diff
@@ -118,4 +113,32 @@
     </div>
 }
 
-@await Html.PartialAsync("Partials/CommentsPartial")
\ No newline at end of file
+@await Html.PartialAsync("Partials/CommentsPartial")
+
+@if (Model.User != null && Model.User.IsAdmin)
+{
+    <div class="ui yellow segment">
+        <h2>Admin Options</h2>
+
+        @if (!Model.ProfileUser.Banned)
+        {
+            <a class="ui red button" href="/admin/user/@Model.ProfileUser.UserId/ban">
+                <i class="ban icon"></i>
+                <span>Ban User</span>
+            </a>
+        <br><br>
+        }
+
+        <form method="post" action="/admin/user/@Model.ProfileUser.UserId/setGrantedSlots">
+            @Html.AntiForgeryToken()
+            <div class="ui left action input">
+                <button type="submit" class="ui blue button">
+                    <i class="pencil icon"></i>
+                    <span>Set Granted Slots</span>
+                </button>
+                <input type="text" name="grantedSlotCount" placeholder="Granted Slots" value="@Model.ProfileUser.AdminGrantedSlots">
+            </div>
+        </form>
```

Comment:
Can this form be turned into a partial so there is no duplicated code between this and `AdminSetGrantedSlotsPage.cshtml`?

Reply:
Probably. Will do so in the next commit.